helsingborg-stad/app-mitt-helsingborg
# Feat/booking summary screen

## Explain the changes you’ve made

I have added a summary screen for showing a successful booking. The same screen will also be used to show more details about bookings in the calendar tab. The feature modal is also updated with the ability to forward navigation params on navigation or changing the content.

## Explain why these changes are made

These screens are part of the booking flow in the app.

## Explain your solution

The FeatureModal component was updated with additional props and the added ability to store and forward navigation params, and the corresponding functions were updated in the modal screens that used navigation.
The summary screen was created using the components from #565.

## How to test the changes?

Confirmation screen:
1. Checkout this branch
2. Add a test button in a screen of your choice, for example `CaseOverview` that navigates to `FeatureModal`
3. Add a new button to Features that changes the modal content to some booking item
4. Fire upp the simulator by running the command `yarn ios`
5. Click on the test button you added in step 2 to summon the modal
6. Click on the new button in the modal to navigate to the confirmation screen

Detail screen:
1. Checkout this branch
2. Make sure the project is not in storybook mode
3. Fire upp the simulator by running the command `yarn ios`
4. Go to the calendar tab
5. Click on a booking
6. You should now be navigated to a detail screen

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
